src/libCli/Completion.cpp

```diff
@@ -19,198 +19,242 @@ using namespace ArgParse;
 
 namespace cli
 {
-
-void printFishCompletions( std::vector<std::shared_ptr<ParsedElement> > & f_candidates, ParsedElement & f_parseTree, const std::string & f_args, bool f_debug)
-{
-    // completion requested :)
-    if(f_debug)
-    {
-        std::cerr << "Input string \"" << f_args << std::endl;
-    }
-
-    //size_t n = parseTree.getMatchedString().size();
-    size_t n = f_args.size();
-    for(auto candidate : f_candidates)
+    // This function modifies the candidate string, such that:
+    // - Only the arguments after the user input are counted as suggestion
+    // - Only one suggestion is completed at a a time
+    //   (everything after a whitespace in a suggestion should be anouther suggestion)
```

Comment:
this function is not implemented specifically to whitespace. So this doc is wrong.
(I suggest though modifying the function instead. see comment below)